rescan.go

```diff
@@ -239,7 +239,7 @@ func (s *ChainService) rescan(options ...RescanOption) error {
 		curStamp  waddrmgr.BlockStamp
 	)
 	if ro.startBlock == nil {
-		bs, err := s.BestSnapshot()
+		bs, err := s.BestReadyBlock()
```

Comment:
With this change, I think we actually don't need the waiting logic below. As before this commit, we would get the best absolute height, which would likely be beyond the height of the filter headers when we're doing our initial sync. 

This applies for the case where we don't set a start block only.

Reply:
The assumption is now that filter headers will always lag behind block headers, hence the waiting logic below. 

If we go the route of catching up filter headers before block headers are "done", then this could be significantly simplified I think, and possible remove the waiting logic below completely (since a scan could just start from the best filter header height, and we would be notified about new headers as connected blocks).

Reply:
One way to get rid of the waiting logic, is to require the caller to start the rescan at a "known" block, a  block that is not beyond the chaintip returned by `BestReadyBlock`. Then we could just return an error in this case. It depends on the assumption though, currently the scan will start at genesis it looks like if a caller passes in an unknown block.